### PR TITLE
[JENKINS-41881] Automated test to cover fix for JENKINS-40224

### PR DESCRIPTION
**Pull request by varyvol**
[JENKINS-41881](https://issues.jenkins-ci.org/browse/JENKINS-41881)

Automated test that failed before the fix and passes after it.

@reviewbybees @i386 @jglick

### CODE REVIEW

**Comment by armfergom** [comment]
🐝

File: src/test/java/hudson/tasks/MailerTest.java [hunk 1]
```diff
@@ -283,13 +298,21 @@ TestProject buildAndCheck(int expectedSize, String recipient) throws Exception {
...
     }
 
     private final class TestBuild {
+        private final FreeStyleProject project;
         private final FreeStyleBuild build;
         private final String log;
 
-        TestBuild(FreeStyleBuild build) throws Exception {
+        TestBuild(FreeStyleProject project, FreeStyleBuild build) throws Exception {
+            this.project = project;
```
Comment (jglick) [inline src/test/java/hudson/tasks/MailerTest.java]: Unnecessary, just use `build.getProject()`.

**Comment by i386** [comment]
Thanks so much for this @varyvol. LGTM 👍

File: src/test/java/hudson/tasks/MailerTest.java (hunk 1 above)
Comment (varyvol) [inline src/test/java/hudson/tasks/MailerTest.java]: @jglick changed.

**Comment by recena** [comment]
🐝

**Comment by varyvol** [comment]
@i386 would you merge this? I think it is ready.

**Pull request merged**